sky/packages/sky/lib/src/rendering/box.dart

```diff
@@ -23,7 +23,22 @@ class _DebugSize extends Size {
   final bool _canBeUsedByParent;
 }
 
+/// Immutable layout constraints for box layout
+///
+/// A size respects a BoxConstraints if, and only if, all of the following
+/// relations hold:
+///
+/// * `minWidth <= size.width && size.width <= constraints.maxWidth`
+/// * `minHeight <= size.height && size.height <= maxHeight`
+///
+/// The box constraints itself must satisfy these relations:
+///
+/// * `0.0 <= minWidth <= maxWidth <= double.INFINITY`
+/// * `0.0 <= minHeight <= maxHeight <= double.INFINITY`
+///
+/// Note: `double.INFINITY` is a legal value for each constraint.
 class BoxConstraints extends Constraints {
+  /// Constructs a box constraints with these constraints
```

Comment:
ditto for subsequent comments. "box constraints" is plural, "BoxConstraints" is singular.